app/assets/javascripts/discourse/app/widgets/topic-admin-menu.js

```diff
@@ -158,7 +158,12 @@ export default createWidget("topic-admin-menu", {
           label: "actions.recover"
         });
       }
+    }
 
+    if (
+      this.currentUser &&
+      (this.currentUser.get("canManageTopic") || details.get("can_close_topic"))
```

Comment:
Do we still need the user and `canManageTopic` checks here if we can check `can_close`, `can_archive`, etc?

Reply:
You're right, canManageTopic is redundant when used alongside can_close_topic or can_archive_topic. I will remove them.